suitcase/utils/conftest.py

```diff
@@ -0,0 +1,130 @@
+from bluesky.tests.conftest import RE # noqa
+from ophyd.tests.conftest import hw # noqa
+from . import SuitcaseUtilsUnknownEventType
+import event_model
+from bluesky.plans import count
+import pytest
+
+
+@pytest.fixture(params=['event', 'bulk_events', 'event_page'],
+                scope='function')
+def image_data(RE, hw, request):
+    return  events_data(RE, hw, det='direct_img', event_type=request.param)
+
+@pytest.fixture(params=['event', 'bulk_events', 'event_page'],
+                scope='function')
+def scalar_data(RE, hw, request):
+    return events_data(RE, hw, det='det', event_type=request.param)
+
+def events_data(RE, hw, det='det', event_type='event'):
+    '''Generates data to be used for testing of suitcase.*.export(..) functions
+
+    Parameters
+    ----------
+    RE : object
+        bluesky.RunEngine object to be used to generate the data.
+    hw : object
+        ophyd.sim.hw object from which to extract simulated detectors and
+        motors.
+    det : string
+        The name of the detector to use to generate the data.
+    event_type : string
+        The event type to return (can be 'event', 'bulk_events' or
+        'event_page')
+    Returns
+    -------
+    collector : list
+        A list of (name, doc) tuple pairs generated by the run engine.
+    '''
+
+    # define the output lists and an internal list.
+    collector = []
+    event_list = []
+
+    # define the collector function depending on the event_type
+    if event_type == 'event':
```

Comment:
Why not simply collect all the documents? That seems like it would be a lot simpler. Then, in `create_expected`, you can loop over the list and select only the event variants.

Reply:
I agree with Dan, having something that returns all of the documents and then picking out what you want in the tests seems much simpler.  I would have expected this to look something like https://github.com/NSLS-II/bluesky/blob/7eea5f30d5e3a4c386585e23d17048b16cdac776/bluesky/tests/utils.py#L30-L46 but handling the extra documents.

Reply:
I am going to push back again, 

`create_expected` can not take in `collector` without adding  lot of complexity to first search through and extract out `event`, `bulk_events` or `event_page` documents. It would then need to convert these to a list of `event` documents (or one of the other types) before starting to collate the information into the expected dictionary.

It makes _no sense_ to me to  do this when I have already created the list of events that I need in `events_data`. I see it as adding extra unnecessary code.

Reply:
> Probably also makes sense to parameterize this by plan so as we add test cases they fan out nicely.

Good idea.

One way to illustrate the limits of this custom data structure is to use a plan that has two EventDescriptors for the 'primary' stream.

```py
@bluesky.preprocessors.run_decorator({})
def plan():
    yield from trigger_and_read(det)
    yield from configure(det, {}, {})  # something like that, see docstring
    yield from trigger_and_read(det)
```

and of course one that has two streams:

```py
@bluesky.preprocessors.baseline_decorator([det])
def plan():
    yield from count([det])